### ISSUE TITLE
V3 - act like protocol buffer

### ISSUE THREAD

**Issue by MirS0bhan** [pydantic/pydantic#10216]
hey fellas
i explain this with 3 indicates : why, what and how 

###  WHY
pydantic is so capable in many usages. also we have serialization delay  and problems. if pydantic being able in v3 to make a property in models to be able for transmission in network without transforming to json or plain text it makes it more powerful than before. 
i love that when you admit this feature and I yell and declare war to google LOL.

### WHAT
first we need to distinct Bufferable Models  of norm models then we need two function buff_stream and buff_loader 
with this logic we would not need any serialization or deserialization. 

### HOW
i don't know. bring your ideas.

**Comment by sydney-runkle** [comment]
Hi @MirS0bhan,

Thanks for the _creative_ feature request. One thing that we've been thinking about recently is a sort of lazy model approach, explained by @samuelcolvin here: https://github.com/pydantic/pydantic/issues/10032

What are your thoughts on how that intersects with your feature request?

**Comment by MirS0bhan** [comment]
@sydney-runkle , @samuelcolvin i read it and i think it is more related to my prior issue #10089.  in that case i prefer that we divide Types into Normally types that we have and Lazy types. pydantic core should ignore them from Model construction until user want to  dump the model and then Lazy types would get validated there. but i think my currently issue is a kinda different concept. Lazy validation is not a way to avoid the overhead of serialization. protocol buffer bring a fast way to avoid that.

**Comment by samuelcolvin** [comment]
I think there's a strong argument to support cbor serialization and deserialization, which should be significantly faster than JSON.

I can give you a list of reasons why protobuf/capnproto or one of the new rust known-schema-(de)serialisation formats would have a poor developer experience, can you tell me why they would be any better than cbor (except performance which I think would be pretty similar in absolute terms).

---

_(@MirS0bhan side note: please can you try to explain your ideas with technical details — enthusiasm is great, but detail and specificity go a long way in conversations like this 🙏)_

**Comment by MirS0bhan** [comment]
@samuelcolvin are you using the cbor already in pydantic core ?

**Comment by samuelcolvin** [comment]
no, we'd need to add it, but if we added it to Jiter correctly, it might require minimal extra logic in pydantic-core.

**Comment by MirS0bhan** [comment]
@samuelcolvin in fact i didn't mean exactly implement  protocol buffer. I'm not senior and I didn't know a lot about these option that's we have. so protobuf just was a sample of what i want to mean. so yeah, it was so inspiring as when we doesn't have any overhead of (de)serialization. also I'm attempt to create a framework for operational system and not Informational systems like django. and I see it is look like RPC and gRPC powers came from the protobuf. and also i use pydantic for manage my operations data as pydantic data classes for better management. so in that state framework would gather all operation by sessions and send them to services, here I would create a gap and send operation to services over the network. and here we have overhead of pydantic validation and (de)serialization. at the final i came here to say pydantic could be more powerful by implementing this feature.
